docs/publish/deploy-office-add-in-sso-to-azure.md

```diff
@@ -0,0 +1,253 @@
+      console.log("static set up: " + path.join(process.cwd()));
+    }
+    
+    const indexRouter = express.Router();
+    indexRouter.get("/", function (req, res) {
+      //   res.render("/taskpane.html");
+      res.sendFile("/taskpane.html", { root: __dirname });
+    });
+    
+    app.use("/", indexRouter);
+    
+    app.get("/getuserdata", validateJwt, getUserData);
+    
+    // Catch 404 and forward to error handler
+    app.use(function (req, res, next) {
+      next(createError(404));
+    });
+    
+    // error handler
+    app.use(function (err, req, temp, res) {
+      // set locals, only providing error in development
+    
+      res.locals.message = err.message;
+      res.locals.error = req.app.get("env") === "development" ? err : {};
+    
+      // render the error page
+      res.status(err.status || 500).send({
+        message: err.message,
+      });
+    });
+    
+    app.listen(process.env.PORT, () => console.log("Server listening on port: " + process.env.PORT));
+    ```
+
+1. Save the file.
+
+## Update app registration
+
+We recommend you create multiple app registrations for localhost, staging, and deployment testing. The following steps ensure that the app registration you use for deployment correctly uses the app service URL.
+
+1. In the Azure portal, open your app registration. Note that the app registration may be in a different account than your app service. Be sure to sign in to the correct account.
+1. In the left sidebar, select **Authentication**.
+1. On the **Authentication** pane, find the `https://localhost:3000/fallbackauthdialog.html` and change it to use the app service URL you saved previously. For example, `https://contoso.sso.azurewebsites.net/fallbackauthdialog.html`.
+1. Save the change.
+1. In the left sidebar, select **Expose an API**.
+1. Change the **Application ID URI** to use your app service URL you saved previously. For example, `api://contoso-sso.azurewebsites.net/628050c7-8d46-4f8f-a393-ac22eb688477`.
+1. Save the changes.
+
+## Build and deploy
+
+Once the files and app registration are updated, you can deploy the add-in.
+
+1. In VS Code open the terminal and run the command `npm run build`. This will build a folder named `dist` that you can deploy.
+1. In the VS Code **Explorer** browse to the `dist` folder. Right-click the `dist` folder and select **Deploy to Web App..**.
+1. When prompted to select a resource, select the app service you created previously.
+1. When prompted if you are sure, select **Deploy**.
+1. When prompted to always deploy the workspace, choose **Yes**.
+
+If you make additional code change, you will need to run `npm run build` again and redeploy the project.
+
+## Test the deployment
+
+Sideload the **manifest-deployment.xml** and test the functionality of the add-in in Office. For more information, see [Sideload an Office Add-in for testing](../testing/test-debug-office-add-ins.md#sideload-an-office-add-in-for-testing).
```

Comment:
I ended up getting this error, which is probably because the Expose an API step failed.

"AADSTS500011: The resource principal named api://aljerabe-deploytest.azurewebsites.net/68554f15-e421-4b3a-9e8f-7541bb7191eb was not found in the tenant named 36fde7ea-8595-437c-aed6-60c976d36083. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.\r\nTrace ID: 0dfe38e3-425b-4094-841a-fcecabaa0700\r\nCorrelation ID: 2e781606-2cef-437c-942e-1679e5ed0f20\r\nTimestamp: 2022-12-28 22:54:40Z"

Reply:
Yep, most likely the api: https: mismatch.